typescript/cli/src/deploy/warp.ts

```diff
@@ -128,10 +151,110 @@ async function executeDeploy(params: DeployParams) {
     log('Writing deployment artifacts');
     await registry.addWarpRoute(warpCoreConfig);
   }
-  log(JSON.stringify(warpCoreConfig, null, 2));
+  log(indentYamlOrJson(yamlStringify(warpCoreConfig, null, 2), 4));
   logBlue('Deployment is complete!');
 }
 
+async function deployAndResolveWarpIsm(
+  warpConfig: WarpRouteDeployConfig,
+  multiProvider: MultiProvider,
+  registry: IRegistry,
+  ismFactoryDeployer: HyperlaneProxyFactoryDeployer,
+): Promise<WarpRouteDeployConfig> {
+  return promiseObjAll(
+    objMap(warpConfig, async (chain, config) => {
+      // Skip deployment if Ism is empty, or a string
+      if (
+        !config.interchainSecurityModule ||
+        typeof config.interchainSecurityModule === 'string'
+      ) {
+        logGray(
+          `Config Ism is ${
+            !config.interchainSecurityModule
+              ? 'empty'
+              : config.interchainSecurityModule
+          }, skipping deployment`,
+        );
+        return config;
+      }
+
+      logBlue('Loading Registry factory addresses');
+      let chainAddresses = await registry.getChainAddresses(chain); // Can includes other addresses
+
+      if (!chainAddresses) {
+        logGray('Registry factory addresses not found, deploying');
+        chainAddresses = serializeContracts(
+          await ismFactoryDeployer.deployContracts(chain),
```

Comment:
is this something we'll always want to deploy all at once? is there a world with lazy factory deploys or not a big deal?